src/amqp-base-client.ts

```diff
@@ -46,6 +47,7 @@ export abstract class AMQPBaseClient {
     })
     if (name) this.name = name // connection name
     if (platform) this.platform = platform
+    if (logger !== undefined) this.logger = logger
```

Comment:
```suggestion
    this.logger = logger
```

We already support it being undefined so the code must handle that anyway.